com.civicrm.hrjobroles/js/src/job-roles/controllers/hr-job-roles-controller.js

```diff
@@ -998,6 +997,18 @@ define([
             function getContractData(contractId) {
               return me.contractsData[contractId];
             }
+
+            /**
+             * Parse dates so they can be correctly read by server.
+             *
+             * @param {string|Date} date
+             * @returns {string|null}
+             */
+            function convertDateToServerFormat(date) {
```

Comment:
Tbh I wanted `$filter('formatDate')` to return `null` or `undefined` when the date wasn't valid or set, but apparently some applications rely on the specific `Unspecified` string. This function was a simple way to make things still work without changing the response from the filter (and then introducing some more breaking changes), but if there was some work to be put to get a more common logic, it wouldn't be to reuse this function, but to amend the filter to *not* return `Unspecified`, and then refactor the places where we rely on that string. Given the scope of this PR I had to give up on that idea (for now!)